cmd/soroban-cli/src/commands/contract/info/meta.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::Debug;
+
+use crate::commands::contract::info::meta::Error::{NoMetaPresent, NoSACMeta};
+use crate::commands::contract::info::shared;
+use crate::commands::contract::info::shared::fetch_wasm;
+use clap::{command, Parser};
+use soroban_spec_tools::contract;
+use soroban_spec_tools::contract::Spec;
+use stellar_xdr::curr::{ScMetaEntry, ScMetaV0};
+
+use crate::commands::contract::InfoOutput;
+
+#[derive(Parser, Debug, Clone)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub common: shared::Args,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Wasm(#[from] shared::Error),
+    #[error(transparent)]
+    Spec(#[from] contract::Error),
+    #[error("Stellar asset contract doesn't contain meta information")]
+    NoSACMeta(),
+    #[error("no meta present in provided WASM file")]
+    NoMetaPresent(),
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<String, Error> {
+        let bytes = fetch_wasm(&self.common).await?;
+
+        if bytes.is_none() {
+            return Err(NoSACMeta());
+        }
+        let spec = Spec::new(&bytes.unwrap())?;
+
+        if spec.meta_base64.is_none() {
+            return Err(NoMetaPresent());
+        }
+
+        let res = match self.common.output {
+            InfoOutput::XdrBase64 => spec.meta_base64.unwrap(),
+            InfoOutput::Json => serde_json::to_string(&spec.meta)?,
+            InfoOutput::JsonFormatted => serde_json::to_string_pretty(&spec.meta)?,
+            InfoOutput::Pretty => {
+                let mut meta_str = "Contract meta:\n".to_string();
+
+                for meta_entry in &spec.meta {
+                    match meta_entry {
+                        ScMetaEntry::ScMetaV0(ScMetaV0 { key, val }) => {
+                            let key = key.to_string();
+                            let (key, val) = match key.as_str() {
+                                "rsver" => ("Rust version", val.to_string()),
+                                "rssdkver" => (
+                                    "Soroban SDK version",
+                                    format!("{})", val.to_string().replace('#', " (commit hash: ")),
+                                ),
+                                _ => (key.as_str(), val.to_string()),
+                            };
+                            meta_str.push_str(&format!(" • {key}: {val}\n"));
```

Comment:
I like the UX the mapping creates, but I think we should remove the mapping and output the raw keys and values, mostly because devs will be able to figure it out, and it gives them the same 1-1 experience of the meta that they'll have if they write any tools to process the values.

Reply:
Don't mind just outputting the raw keys, but I'd say we should aim towards people not using pretty output in any tools. We have json format for that (that's much easier to use in tools), that will print the same output in between minor versions. For pretty printing, I think we can (and should be able to) break it without a notice in a minor version

Reply:
I'm less concerned with breaking the output. I agree the pretty / text output shouldn't provide many guarantees on content or structure.

I'm more concerned with creating a consistent narrative. For example, someone views a contract with a mix of custom keys plus the common rsver and rssdk. They'd see output like this:
```
$ stellar-cli contract info meta --wasm contract.wasm 
Contract meta:
 • Rust version: 1.79.0
 • Soroban SDK version: 21.4.0 (commit hash: d6f5639f643d76e758beecbb0ca391f8cd304c24)
 • customfield: myvalue
```

A developer may be mistaken to think they can lookup the outputted values using the keys `customfield`, `Rust version`, `Soroban SDK Version`. When they actually need to lookup using keys `customfield`, `rsver`, `rssdk`.

We could achieve key consistency and educate the user with less pretty but just as functional output:

```
$ stellar-cli contract info meta --wasm contract.wasm 
Contract meta:
 • rsver: 1.79.0 (Rust Version)
 • rssdk: 21.4.0#d6f5639f643d76e758beecbb0ca391f8cd304c24 (Rust SDK Version)
 • customfield: myvalue
```

Reply:
Oh yeah I like that idea!